src/Transport.ts

```diff
@@ -331,320 +338,383 @@ export default class Transport {
       }
-    }
 
-    headers.accept = headers.accept ?? this[kAcceptHeader]
-    connectionParams.headers = headers
-    while (meta.attempts <= maxRetries) {
-      try {
-        if (signal?.aborted) { // eslint-disable-line
-          throw new RequestAbortedError('Request has been aborted by the user', result, errorOptions)
-        }
+      headers.accept = headers.accept ?? this[kAcceptHeader]
+      connectionParams.headers = headers
+      while (meta.attempts <= maxRetries) {
+        try {
+          if (signal?.aborted) { // eslint-disable-line
+            throw new RequestAbortedError('Request has been aborted by the user', result, errorOptions)
+          }
 
-        meta.connection = this.getConnection({
-          requestId: meta.request.id,
-          context: meta.context
-        })
-        if (meta.connection === null) {
-          throw new NoLivingConnectionsError('There are no living connections', result, errorOptions)
-        }
+          meta.connection = this.getConnection({
+            requestId: meta.request.id,
+            context: meta.context
+          })
+          if (meta.connection === null) {
+            throw new NoLivingConnectionsError('There are no living connections', result, errorOptions)
+          }
 
-        this[kDiagnostic].emit('request', null, result)
-
-        // perform the actual http request
-        let { statusCode, headers, body } = await meta.connection.request(connectionParams, {
-          requestId: meta.request.id,
-          name: this[kName],
-          context: meta.context,
-          maxResponseSize,
-          maxCompressedResponseSize,
-          signal,
-          timeout: toMs(options.requestTimeout != null ? options.requestTimeout : this[kRequestTimeout]),
-          ...(options.asStream === true ? { asStream: true } : null)
-        })
-        result.statusCode = statusCode
-        result.headers = headers
-
-        if (this[kProductCheck] != null && headers['x-elastic-product'] !== this[kProductCheck] && statusCode >= 200 && statusCode < 300) {
-          /* eslint-disable @typescript-eslint/prefer-ts-expect-error */
-          // @ts-ignore
-          throw new ProductNotSupportedError(this[kProductCheck], result, errorOptions)
-          /* eslint-enable @typescript-eslint/prefer-ts-expect-error */
-        }
+          // generate required OpenTelemetry attributes from the request URL
+          const requestUrl = meta.connection.url
+          otelSpan?.setAttributes({
+            'url.full': requestUrl.toString(),
+            'server.address': requestUrl.hostname
+          })
+          if (requestUrl.port === '') {
+            if (requestUrl.protocol === 'https:') {
+              otelSpan?.setAttribute('server.port', 443)
+            } else if (requestUrl.protocol === 'http:') {
+              otelSpan?.setAttribute('server.port', 80)
+            }
+          } else if (requestUrl.port !== '9200') {
```

Comment:
I'm curious. Why port "9200" is omitted?

Reply:
@david-luna I think it is a spec thing. A note on the `server.port` field at https://opentelemetry.io/docs/specs/semconv/database/elasticsearch/ is:

> [7]: If using a port other than the default port for this DBMS and if server.address is set.

And 9200 is the default ES port.